internal/credential/verification.go

```diff
@@ -0,0 +1,176 @@
+package credential
+
+import (
+	"crypto"
+	"fmt"
+
+	credsdk "github.com/TBD54566975/ssi-sdk/credential"
+	"github.com/TBD54566975/ssi-sdk/credential/signing"
+	"github.com/TBD54566975/ssi-sdk/credential/verification"
+	"github.com/TBD54566975/ssi-sdk/cryptosuite"
+	didsdk "github.com/TBD54566975/ssi-sdk/did"
+	"github.com/goccy/go-json"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/mr-tron/base58"
+	"github.com/multiformats/go-multibase"
+	"github.com/multiformats/go-varint"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/tbd54566975/ssi-service/internal/keyaccess"
+	"github.com/tbd54566975/ssi-service/internal/util"
+)
+
+type CredentialVerifier struct {
+	verifier *verification.CredentialVerifier
+	resolver *didsdk.Resolver
+}
+
+// NewCredentialVerifier creates a new credential verifier which executes both signature and static verification checks.
+// In the future the set of verification checks will be configurable.
+func NewCredentialVerifier(resolver *didsdk.Resolver) (*CredentialVerifier, error) {
```

Comment:
eventually we will get a whole test suit setup with a lot of real deal signed creds and verify theses functions

Reply:
agreed

Reply:
agree